### DISCUSSION TITLE
Retrieve Build Number of a Build (Workflow and Application Scope)

### DISCUSSION THREAD

**Discussion by nevotheless** [codemagic-ci-cd/codemagic-docs#2188] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

```
curl --header "x-auth-token: ${someToken}" https://api.codemagic.io/builds/${someBuildID}
```

The resulting output doesn't contain any info about build numbers.

### Expected results

I would like to be able to get Build Numbers from Workflow and Project/Application Scope.

### Actual results

No Build Number Information via the Builds API even though the data is probably easily available (as seen in the codemagic website).

**Comment by nevotheless** [comment]
I have been able to workaround this issue using a custom version file that also included the build numbers i needed and exporting it as an artifact using the `$CM_EXPORT_DIR` mechanism. But maybe exposing the build numbers on the API would still be a good idea?

**Reply by dtrdic** [reply, marked as answer]
hey @nevotheless, i'm glad you found a workaround. Just a note that our current API is about to be depreciated and we are working to release a new improved version, and will certainly have this kind of information
